billing-service/src/main/java/com/petclinic/billing/presentationlayer/BillResource.java

```diff
@@ -32,10 +32,8 @@ public Mono<ResponseEntity<BillResponseDTO>> createBill(@Valid @RequestBody Mono
 
     // Read Bill //
     @GetMapping(value = "/bills/{billId}")
-    public Mono<ResponseEntity<BillResponseDTO>> findBill(@PathVariable("billId") String billId){
-        return SERVICE.GetBill(billId)
-                .map(ResponseEntity::ok)
-                .defaultIfEmpty(ResponseEntity.notFound().build());
+    public Mono<BillResponseDTO> getBillByBillId(@PathVariable String billId){
+        return SERVICE.getBillByBillId(billId);
```

Comment:
How come we're losing the response entity here?

Reply:
I think I might've not seen that change from a merge conflict. I will add it back as a bug fix